### DISCUSSION TITLE
Output Derivatives in Inverse Problem!!!!

### DISCUSSION THREAD

**Discussion by ppandit95** [lululxvi/deepxde#967] [Q&A] [answered]
Dear Lu Lu

Firstly, thanks alot for bringing in this open source library and it works great.....

Secondly,I am trying to solve inverse problem involving simple Fick's First Law of Diffusion wherein I wish to get the values of derivatives at each training point in post processing so that I can verify my simulation and from one of the examples I implemented it as 
`
def get_derivatives(x,M):
        dM1_x = dde.grad.jacobian(M,x,i=0,j=0)
        dM2_x = dde.grad.jacobian(M,x,i=1,j=0)
        return [dM1_x,dM2_x]
`
After training the model I called these below lines -
`
f=model.predict(observe_x,operator = get_derivatives)
np.savetxt("Derivatives.dat",np.array(f))
`

But upon running I am getting an output as
`
Traceback (most recent call last):
  File "hPINN-Alternative_not_working.py", line 268, in <module>
    np.savetxt(f"Derivatives.dat",[f[0],f[1]])
  File "<__array_function__ internals>", line 180, in savetxt
  File "/home/cmslab/.local/lib/python3.8/site-packages/numpy/lib/npyio.py", line 1507, in savetxt
    raise ValueError(
ValueError: Expected 1D or 2D array, got 3D array instead
`
Any help in this regard will prove beneficial.

many Thanks
Pushkar

**Comment by tsarikahin** [comment, marked as answer]
The problem here is that np.savetxt. Check here https://stackoverflow.com/questions/51218937/saving-a-3d-numpy-array-to-txt-file.
I think this should work
```python
def get_derivatives(x,M): 
      dM1_x = dde.grad.jacobian(M,x,i=0,j=0) 
      dM2_x = dde.grad.jacobian(M,x,i=1,j=0) 
      return dM1_x,dM2_x

f=model.predict(observe_x,operator = get_derivatives)
res_stacked = np.hstack((observe_x.reshape(-1,1),f[0].reshape(-1,1),f[1].reshape(-1,1)))
np.savetxt("Derivatives.dat",res_stacked)
```

**Reply by ppandit95** [reply]
Thanks @tsarikahin for helping as I am able to get the derivatives ......
